src/overlays/actors/ovl_En_Aob_01/z_en_aob_01.c

```diff
@@ -42,92 +56,1011 @@ const ActorInit En_Aob_01_InitVars = {
+                this->actionFunc = func_809C2BE4;
+            }
+        } else if (gSaveContext.weekEventReg[8] & 0x20) {
+            func_800B8A1C(&this->actor, globalCtx, 4, 300.0f, 300.0f);
+        } else {
+            func_800B8A1C(&this->actor, globalCtx, 12, 300.0f, 300.0f);
+        }
+    }
+}
+
+void func_809C2BE4(EnAob01* this, GlobalContext* globalCtx) {
+    u8 temp_v0 = func_80152498(&globalCtx->msgCtx);
+
+    if (((temp_v0 == 5) || (temp_v0 == 6)) && func_80147624(globalCtx)) {
+        if (gSaveContext.weekEventReg[63] & 2) {
+            gSaveContext.weekEventReg[63] &= (u8)~2;
+        }
+
+        if (gSaveContext.weekEventReg[63] & 1) {
+            gSaveContext.weekEventReg[63] &= (u8)~1;
+        }
+
+        this->unk_210 = 0;
+        func_800B85E0(&this->actor, globalCtx, 400.0f, -1);
+        gSaveContext.eventInf[0] &= (u8)~7;
+        this->actionFunc = func_809C2C9C;
+    }
+}
+
+void func_809C2C9C(EnAob01* this, GlobalContext* globalCtx) {
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        this->unk_210 = 0x354C;
+        func_80151938(globalCtx, this->unk_210);
+        this->actionFunc = func_809C1D64;
+    } else {
+        func_800B85E0(&this->actor, globalCtx, 400.0f, -1);
+    }
+}
+
+void func_809C2D0C(EnAob01* this, GlobalContext* globalCtx) {
+    u8 sp2F = func_80152498(&globalCtx->msgCtx);
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 4, 4000, 1);
+
+    if (this->unk_2D2 & 0x40) {
+        if (!func_809C1424(this)) {
+            return;
+        }
+    } else if (this->unk_2D2 & 0x80) {
+        if (!func_809C14D0(this)) {
+            return;
+        }
+    }
+
+    if ((sp2F == 5) && func_80147624(globalCtx)) {
+        this->unk_2D2 &= ~0x40;
+        this->unk_2D2 &= ~0x80;
+        if (this->unk_434 >= 150) {
+            this->unk_210 = 0x352E;
+            func_801518B0(globalCtx, this->unk_210, &this->actor);
+            this->actionFunc = func_809C2A64;
+        } else {
+            this->unk_2D4 = this->unk_2E6;
+            this->unk_2DA = this->unk_2EC;
+            this->unk_2E0 = this->unk_2F2;
+
+            this->unk_434 = 0;
+            this->actor.shape.rot.y = this->actor.world.rot.y;
+            if (gSaveContext.weekEventReg[63] & 2) {
+                gSaveContext.weekEventReg[63] &= (u8)~2;
+            }
+
+            if (gSaveContext.weekEventReg[63] & 1) {
+                gSaveContext.weekEventReg[63] &= (u8)~1;
+            }
 
-extern ColliderCylinderInit D_809C3820;
+            this->unk_210 = 0x354C;
+            func_801518B0(globalCtx, this->unk_210, &this->actor);
+            this->actionFunc = func_809C1D64;
+            gSaveContext.eventInf[0] &= (u8)~7;
+        }
+    }
+}
 
-extern UNK_TYPE D_06000180;
-extern UNK_TYPE D_06003D18;
+s32 func_809C2EC4(EnAob01* this, GlobalContext* globalCtx) {
+    Actor* doggywoggy = globalCtx->actorCtx.actorList[ACTORCAT_ENEMY].first;
```

Comment:
```suggestion
    Actor* dog = globalCtx->actorCtx.actorList[ACTORCAT_ENEMY].first;
```
Try to avoid such informal names in the repo